src/functional/ffront/foast_passes/type_deduction.py

```diff
@@ -37,17 +39,37 @@ def are_broadcast_compatible(left: TypeInfo, right: TypeInfo) -> bool:
     >>> are_broadcast_compatible(int_field_t, int_scalar_t)
     True
 
+    >>> from functional.iterator.runtime import CartesianAxis
+    >>> Edge = CartesianAxis("Edge")
+    >>> K = CartesianAxis("K")
+    >>> are_broadcast_compatible(
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[K])),
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[Edge, K])),
+    ... )
+    True
+
+    >>> are_broadcast_compatible(
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[Edge])),
+    ...     TypeInfo(ct.FieldType(dtype=ct.ScalarType(kind=ct.ScalarKind.INT64), dims=[Edge, K])),
+    ... )
+    True
     """
     both_dims_given = bool(left.dims and right.dims)
     both_dims_given &= not isinstance(left.dims, GenericDimensions)
     both_dims_given &= not isinstance(right.dims, GenericDimensions)
-    if both_dims_given and any(
-        ldim != rdim
-        for ldim, rdim in zip(
-            left.dims, right.dims  # type: ignore[arg-type]  # we know they are lists here
+    if (
+        both_dims_given and isinstance(left.dims, list) and isinstance(right.dims, list)
+    ):  # isinstance calls are for mypy
+        smaller_dims, bigger_dims = (
+            (left.dims, right.dims)
+            if len(left.dims) <= len(right.dims)
+            else (right.dims, left.dims)
         )
-    ):
-        return False
+        if smaller_dims[0] in bigger_dims and smaller_dims[-1] in bigger_dims:
+            start_index = bigger_dims.index(smaller_dims[0])
+            end_index = bigger_dims.index(smaller_dims[-1])
+            if smaller_dims != bigger_dims[start_index : end_index + 1]:
+                return True
```

Comment:
```suggestion
                return False
```
If the two don't match it should return `False` right?